use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,484 @@
+```
+
+### Copy Kata rootfs
+
+```sh
+$ mkdir -p $KATA_ROOTFS_LOCATION
+$ cp ${OSBUILDER}/osbuilder/image-builder/kata-containers.img $KATA_ROOTFS_LOCATION
+```
+
+### Update Kata configuration to point to custom kernel and rootfs
+
+You must update the `configuration.toml` for Kata Containers to point to the 
+custom kernel, custom rootfs, and to specify which modules to load when the 
+virtual machine is booted when a container is run. The example below assumes 
+an Intel QAT is installed and those modules need to be loaded.
+
+```sh
+$ sudo mkdir -p /etc/kata-containers
+$ sudo cp /usr/share/defaults/kata-containers/configuration-qemu.toml /etc/kata-containers/configuration.toml
+
+# Make sure usdm_drv is loaded in the image
+$ sudo sed -i "s|kernel_params = \"\"|kernel_params = \"modules-load=usdm_drv,qat_c62xvf\"|g" /etc/kata-containers/configuration.toml
+
+# Change the default kernel and rootfs
+$ sudo sed -i "s|\/usr\/share\/kata-containers\/kata-containers.img|${KATA_KERNEL_LOCATION}\/kata-containers.img|g" /etc/kata-containers/configuration.toml
+$ sudo sed -i "s|\/usr\/share\/kata-containers\/vmlinuz.container|${KATA_ROOTFS_LOCATION}\/vmlinuz-${LINUX_VER}_qat|g" /etc/kata-containers/configuration.toml
+```
+
+## Verify Intel QAT works in a Docker Kata Containers container
+
+The following instructions leverage an OpenSSL Dockerfile that builds the 
+Intel QAT engine to allow OpenSSL to offload crypto functions. It is a 
+convenient way to test that VFIO device passthrough for the Intel QAT VF’s are
+working properly with the Kata Containers VM.
+
+## Build OpenSSL Intel QAT engine container
+
+There is an OpenSSL Intel QAT Dockerfile located at 
+[https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine](https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine). 
+Use this file to build a container image with an optimized OpenSSL engine for 
+Intel QAT. Using Docker build with the kata runtime can sometimes have issues. 
+Therefore, it is recommended to change the default runtime to runc before 
+doing a build. Instructions for this are below.
+
+
+> **Note: The Intel QAT driver version in this container might not match the 
+> Intel QAT driver compiled and loaded on the host when compiling.**
+
+```sh
+$ cd $QAT_SRC
+$ curl -O $QAT_DOCKERFILE
+$ sudo sed -i 's/kata-runtime/runc/g' /etc/systemd/system/docker.service.d/50-runtime.conf
+$ sudo systemctl daemon-reload && sudo systemctl restart docker
+$ sudo docker build -t openssl-qat-engine .
+```
+
+### Test Intel QAT in Docker
+
+The host should already be setup with 16 virtual functions of the Intel QAT 
+card bound to VFIO-PCI. This can be verified by looking in `/dev/VFIO` for a 
+listing of functions. Replace the number 90 with one of the VF’s exposed 
+in `/dev/VFIO`. It might be required to add an `IPC_LOCK` capability to your 
+Docker runtime depending on which rootfs is used. 
```

Comment:
Lines 402-406 suggested rewrite: 
The host should already be setup with 16 virtual functions of the Intel QAT card bound to VFIO-PCI. Verify this by looking in /dev/VFIO for a listing of functions. Replace the number 90 with one of the VF’s exposed in /dev/VFIO. It might require you to add an IPC_LOCK capability to your Docker runtime depending on which rootfs you use.

Reply:
Done